src/main/java/io/cdap/plugin/successfactors/connector/SuccessFactorsConnector.java

```diff
@@ -0,0 +1,259 @@
+import org.apache.olingo.odata2.api.ep.feed.ODataFeed;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.core.MediaType;
+
+/**
+ * SuccessFactorsConnector Class
+ */
+@Plugin(type = Connector.PLUGIN_TYPE)
+@Name(SuccessFactorsConnector.NAME)
+@Description("Connection to access data in SuccessFactors.")
+public class SuccessFactorsConnector implements DirectConnector {
+  public static final String NAME = "SuccessFactors";
+  public static final String METADATA = "METADATA";
+  public static final String PROPERTY_SEPARATOR = ",";
+  private static final String ENTITY_TYPE_ENTITY = "entity";
+  private static final String METADATACALL = "$metadata";
+  private static final String TOP_OPTION = "$top";
+  private static final String SELECT_OPTION = "$select";
+  private final SuccessFactorsConnectorConfig config;
+
+  public SuccessFactorsConnector(SuccessFactorsConnectorConfig config) {
+    this.config = config;
+  }
+
+
+  @Override
+  public void test(ConnectorContext connectorContext) throws ValidationException {
+    FailureCollector collector = connectorContext.getFailureCollector();
+    config.validateBasicCredentials(collector);
+    config.validateConnection(collector);
+  }
+
+  @Override
+  public BrowseDetail browse(ConnectorContext connectorContext, BrowseRequest browseRequest) throws IOException {
+    BrowseDetail.Builder browseDetailBuilder = BrowseDetail.builder();
+    int count = 0;
+    List<String> entities = null;
+    try {
+      entities = listEntities();
+    } catch (TransportException e) {
+      throw new IOException("Error in communicating SuccessFactors", e);
+    }
+    for (int i = 0; i < entities.size(); i++) {
+      String name = entities.get(i);
+      BrowseEntity.Builder entity = (BrowseEntity.builder(name, name, ENTITY_TYPE_ENTITY).
+        canBrowse(false).canSample(true));
+      browseDetailBuilder.addEntity(entity.build());
+      count++;
+    }
+    return browseDetailBuilder.setTotalCount(count).build();
+  }
+
+  @Override
+  public ConnectorSpec generateSpec(ConnectorContext connectorContext, ConnectorSpecRequest connectorSpecRequest)
+    throws IOException {
+    ConnectorSpec.Builder specBuilder = ConnectorSpec.builder();
+    Map<String, String> properties = new HashMap<>();
+    properties.put(io.cdap.plugin.common.ConfigUtil.NAME_USE_CONNECTION, "true");
+    properties.put(ConfigUtil.NAME_CONNECTION, connectorSpecRequest.getConnectionWithMacro());
+    String entity = connectorSpecRequest.getPath();
+    if (entity != null) {
+      properties.put(SuccessFactorsPluginConfig.ENTITY_NAME, entity);
+    }
```

Comment:
If entity == null, we'll pass null entity to getSchema()

Reply:
reworked on it.